### ISSUE TITLE
Gradle 7.5 Javadoc plugin is broken for multi-module aggregation (due to setting --source-path)

### ISSUE THREAD

**Issue by tcfurrer** [gradle/gradle#21399]
### Expected Behavior
Javadoc task should set only --module-path, and not also --source-path when modularity is enabled.  The --module-source-path switch must be used to properly generate multi-module aggregated javadocs, but is strictly mutex with --source-path.

### Current Behavior
In Gradle 7.4 and earlier, I was able to generate a proper aggregated multi-module javadoc by using:

```
        options.addStringOption('-module-source-path', './*/src/main/java/')
```

This worked because gradle was adding only --module-path (thanks to modularity.inferModulePath).

In Gradle 7.5, the same task fails with:

```
  error: cannot specify both --source-path and --module-source-path
```

This error happens because of a bit of new code which got added in Gradle 7.5, which explicitly sets --source-path for modular projects.  I believe that is incorrect behavior, at least for my project's specific case.

The code I referred to is from this commit:
https://github.com/gradle/gradle/commit/abacb6729a00ea4992341c38909a443a9d52c437

I can manually run javadoc tool successfully by manually removing the --source-path line from the generated javadoc.options file.  However I cannot find any way to easily automate this as a workaround, because Gradle does not have any feature for post-processing of the javadoc.options file for workarounds.

### Context
This regression prevents upgrade beyond Gradle 7.4.

### Steps to Reproduce 
Create a task of type Javadoc which includes sources from multiple modular projects.

### Your Environment
Gradle 7.5

**Comment by tcfurrer** [comment]
The code I referred to is from this commit:

https://github.com/gradle/gradle/commit/abacb6729a00ea4992341c38909a443a9d52c437

**Comment by tcfurrer** [comment]
This is vaguely similar to #2357, except for javadoc instead of javac.

**Comment by tcfurrer** [comment]
As a related note, it would be nice if Gradle could be extended in the future to automatically add --module-source-path as needed for aggregated multi-module javadoc generation.  But in the meantime, it is easy enough to workaround.

**Comment by tcfurrer** [comment]
Also vaguely related: https://github.com/java9-modularity/gradle-modules-plugin/issues/128

**Comment by tcfurrer** [comment]
(Apologies for accidentally filing this as "bug" instead of "regression", since it is clearly a regression.)

**Comment by ljacomet** [comment]
@tcfurrer Any chance you could provide a minimal reproducer that illustrates the problem?

You can use the following as a base for your reproducer: https://github.com/gradle/gradle-issue-reproducer

**Comment by carlosame** [comment]
FWIW the [css4j-dist](https://github.com/css4j/css4j-dist) project is affected by this. With Gradle 7.5:

```shell
$ git clone https://github.com/css4j/css4j-dist.git
$ cd css4j-dist
$ gradle mergedJavadoc
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :mergedJavadoc FAILED
error: cannot specify both --source-path and --module-source-path
1 error

FAILURE: Build failed with an exception.
```

But if you run `mergedJavadoc` with the wrapper (7.4.2 currently) it works.

**Comment by tcfurrer** [comment]
@ljacomet Yes I can work today on providing a minimal reproducer using that template.

**Comment by ljacomet** [comment]
@tcfurrer See the #21442 which is a minimal fix. If you can confirm it works on your project, that would be great. I no longer need a minimal reproducer as I was able to craft a test.

**Comment by tcfurrer** [comment]
Minimal reproducer is here, nonetheless: https://github.com/tcfurrer/gradle-issue-reproducer-21399

**Comment by tcfurrer** [comment]
I grabbed the build artifacts from #21442 and tried them on my original full project. Fix looks good there.

**Comment by ljacomet** [comment]
We decided to  fix this by reverting the fix for #19726 which has a number of issues. See #21446.

**Comment by ljacomet** [comment]
Revert has been merged, behavior in 7.5.1 will match 7.4.2
